data/mods/ssb/moves.ts

```diff
@@ -1616,6 +1616,57 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 		type: "Poison",
 	},
 
+	// WigglyTree
+	perfectmimic: {
+		num: 382,
+		accuracy: true,
+		basePower: 0,
+		category: "Status",
+		isNonstandard: "Past",
+		name: "Perfect Mimic",
+		pp: 10,
+		priority: 0,
+		flags: {protect: 1, bypasssub: 1},
+		volatileStatus: 'perfectmimic',
+		onDisableMove(pokemon) {
+			if (pokemon.lastMove?.id === 'perfectmimic') pokemon.disableMove('perfectmimic');
+		},
+		condition: {
+			duration: 1,
+			onStart(target) {
+				this.add('-singleturn', target, 'move: Perfect Mimic');
+			},
+			onDamagePriority: -10,
+			onDamage(damage, target, source, effect) {
+				if (effect?.effectType === 'Move' && damage >= target.hp) {
+					this.add('-activate', target, 'move: Endure');
+					this.effectState.move = effect;
+					return target.hp - 1;
+				}
+			},
+			onSourceAfterMove(target, source) {
+				if (!source.hp || !this.effectState.move) return;
+				const move = this.effectState.move;
+				const noPerfectMimic = [
+					'beakblast', 'chatter', 'counter', 'covet', 'focuspunch', 'mefirst', 'metalburst', 'mirrorcoat', 'shelltrap', 'struggle', 'thief',
+				];
+				if (move.isZ || move.isMax || move.category === 'Status' || noPerfectMimic.includes(move.id)) return;
```

Comment:
I left the original noMeFirst list here since I didn't know what to remove. Main ones I was confused on were Beak Blast, Focus Punch, and Shell Trap.

Reply:
I don't think you even the me first list, because counter, shell trap, mirror coat, etc. will all fail vs this move.